### PR TITLE
YaruCarousel: fix laggy animation on multiple clicks

### LINKED ISSUE #595: YaruCarousel: click mulltiple times causes laggy animation
[Capture vidéo du 2023-02-05 20-04-30.webm](https://user-images.githubusercontent.com/36476595/216839521-6bf013ad-e8fd-468f-9cf0-76651515896c.webm)

### PR DESCRIPTION
**Pull request by Jupi007**
@jpnurmi I a bit unsure about `return Future.any([]);` what do you think of?

**Before:**

[Capture vidéo du 2023-02-05 20-04-30.webm](https://user-images.githubusercontent.com/36476595/216839521-6bf013ad-e8fd-468f-9cf0-76651515896c.webm)

**After:**

https://user-images.githubusercontent.com/36476595/216843171-60c5a8aa-d870-419b-aee2-b44a214101c5.mp4

Fixes #595

## Pull request checklist

- [x] This PR does not introduce visual changes.

### CODE REVIEW

File: lib/src/widgets/yaru_carousel.dart [hunk 1]
```diff
@@ -244,15 +245,24 @@ class YaruCarouselController extends PageController {
     Duration? duration,
     Curve? curve,
   }) {
```
Comment (jpnurmi) [inline lib/src/widgets/yaru_carousel.dart:247]:
```suggestion
  }) async {
```

File: lib/src/widgets/yaru_carousel.dart [hunk 2]
```diff
@@ -244,15 +245,24 @@ class YaruCarouselController extends PageController {
...
 
-    return super
-        .animateToPage(
-          page,
-          duration: duration ?? scrollAnimationDuration,
-          curve: scrollAnimationCurve,
-        )
-        .then((value) => startTimer());
+    // Return a never finishing future if already animating.
+    return Future.any([]);
```
Comment (jpnurmi) [inline lib/src/widgets/yaru_carousel.dart:265]:
```suggestion
```

**Review by jpnurmi — COMMENTED** [review]
if we mark the method async, we can remove the "void" return value :)

**Comment by Jupi007** [comment]
Thanks a lot!
I was sure it was a dirty fix :D

**Review by jpnurmi — APPROVED** [review]
LGTM :+1:

**Pull request merged**